### DISCUSSION TITLE
Forward search issue from powershell.

### DISCUSSION THREAD

**Discussion by ubaldot** [sumatrapdfreader/sumatrapdf#4532] [Q&A]
I am trying to make it work with Vim, but unfortunately the [guide](https://william.famille-blum.org/blog/static.php?page=static081010-000413) linked in the doc is not reachable.

However, I am positive and I think we can sort it out anyway. :) 

I build the pdf with 

`latexmk -pdf -xelatex -synctex=1 -quiet -interaction=nonstopmode C:\Users\myname\vimfiles\after\example.tex`

So far, so good. I get both the `.pdf` and the `.syntec.gz` file but I have a problem with the forward search. 
 
From a _powershell_ console I run:

`SumatraPDF.exe -reuse-instance -forward-search C:\Users\myname\vimfiles\after\example.tex 10 `

but nothing happens. I also tried to quote the file name but it didn't solve the problem. 

The backwards-search works perfectly!

**Comment by GitHubRulesOK** [comment]
There are several reports of failing with latexmk and seems to be variable per install but that mix of xelatex seems to be the breakage.

That old link is way out of date though it was William introduced SyncTeX to SumatraPDF fork and then mainstream code via DDE.

SumatraPDF works perfectly in all other known mixtures so something is different in the way XeTeX functions which has been potentially addressed but not yet hit the news stands it should be working in Pre-release https://www.sumatrapdfreader.org/prerelease

Please report back if it works as there are multiple unclosed issues without feedback such as https://github.com/sumatrapdfreader/sumatrapdf/issues/2867

P.S. you should not  need that historic `-reuse-instance` it adds an unnecessary call as it should already be be set ***true*** in the settings file however you need to ensure it always works without it :-)

**Comment by ubaldot** [comment]
I actually made it work with Vim, including the backward search, but I am trying to figure out how to make the forward search to work because it is the last thing missing. But for that I left Vim out of the equation and I am studying the problem by calling the command from the powershell, as explained. 

I just tested the prerelease and it does not work. I also tried to build with `pdflatex `instead of `xelatex `and it didn't work either. 
In this case every time I run the command `SumatraPDF.exe -reuse-instance -forward-search C:\Users\myname\vimfiles\after\example.tex 10` it opens a new empty instance of Sumatra.

**Reply by GitHubRulesOK** [reply]
it should not be "empty" perhaps "myname" is not simply  ***ubaldot*** as there can be problems with non unix characters such as a space or an accent?

TABS ON and ReuseInstance = true are basic settings that should not be any problem for a `TeX -forward-search "filename"` in a standard environment.

**Comment by kjk** [comment]
1. You need to provide path to PDF file i.e. `SumatraPDF.exe -forward-search "<path>\foo.tex" <line> "<path>\foo.pdf"`

2. You need to quote paths i.e. `"foo.pdf"`, not just `foo.pdf`, so that it doesn't break when path / file has a space in it

3. You probably need to have SumatraPDF running and PDF file already loaded

4. You probably don't need `-reuse-instance`

5. [Check the logs](https://www.sumatrapdfreader.org/docs/Debugging-Sumatra#getting-logs) if it doesn't work 

`-forward-search` converts the command into DDE command and sends it to existing SumatraPDF instance

`-forward-search` is meant for integration with a text editor. It tells sumatra to go to a position in PDF that matches position in .tex file. So Sumatra needs to know:
* pdf file
* tex file, line
* synctex file to do the mapping

**Reply by ubaldot** [reply]
> 1. You need to provide path to PDF file i.e. `SumatraPDF.exe -forward-search "<path>\foo.tex" <line> "<path>\foo.pdf"`
> 
> 
> 
> 2. You need to quote paths i.e. `"foo.pdf"`, not just `foo.pdf`, so that it doesn't break when path / file has a space in it
> 
> 
> 
> 3. You probably need to have SumatraPDF running and PDF file already loaded
> 
> 
> 
> 4. You probably don't need `-reuse-instance`
> 
> 
> 
> 5. [Check the logs](https://www.sumatrapdfreader.org/docs/Debugging-Sumatra#getting-logs) if it doesn't work 
> 
> 
> 
> `-forward-search` converts the command into DDE command and sends it to existing SumatraPDF instance
> 
> 
> 
> `-forward-search` is meant for integration with a text editor. It tells sumatra to go to a position in PDF that matches position in .tex file. So Sumatra needs to know:
> 
> * pdf file
> 
> * tex file, line
> 
> * synctex file to do the mapping
> 
> 
> 
> 
> 
> 

Thanks! I will try later on today or tomorrow. :)
I used that format because that is what's written in the docs https://www.sumatrapdfreader.org/docs/Command-line-arguments

;-)

**Comment by ubaldot** [comment]
> You need to provide path to PDF file i.e. SumatraPDF.exe -forward-search "<path>\foo.tex" <line> "<path>\foo.pdf"

Tested with the released version of Sumatra and it works perfectly. Many thanks! But please, update the docs. 

P.S. 
If you want I can add few lines how I managed to setup Vim. :)

**Reply by kjk** [reply]
Please do. The docs are in https://github.com/sumatrapdfreader/sumatrapdf/tree/master/docs/md

Best place would be https://github.com/sumatrapdfreader/sumatrapdf/blob/master/docs/md/LaTeX-integration.md 

You can edit it directly in GitHub UI or fork / edit / push / create a PR.
